experimental/frame_sorter.go

```diff
@@ -0,0 +1,46 @@
+package experimental
+
+import "github.com/grafana/grafana-plugin-sdk-go/data"
+
+// NewFrameSorter - returns a new frameSorter
+func NewFrameSorter(frame *data.Frame, sortField *data.Field) FrameSorter {
+	return FrameSorter{frame, sortField}
+}
+
+// FrameSorter - sort a DataFrame by field
+type FrameSorter struct {
+	frame     *data.Frame
+	sortField *data.Field
+}
+
+func (fs FrameSorter) Len() int { return fs.frame.Rows() }
+func (fs FrameSorter) Swap(i, j int) {
+	for _, field := range fs.frame.Fields {
+		valA := field.At(i)
+		valB := field.At(j)
+		field.Set(j, valA)
+		field.Set(i, valB)
+	}
+}
+func (fs FrameSorter) Less(i, j int) bool {
+	switch kind := fs.sortField.Type(); kind {
+	case data.FieldTypeString:
+		valA := fs.sortField.At(i).(string)
+		valB := fs.sortField.At(j).(string)
+		return valA < valB
+	case data.FieldTypeNullableString:
+		valA := fs.sortField.At(i).(*string)
+		valB := fs.sortField.At(j).(*string)
+		return *valA < *valB
+	default:
+		valA, err := fs.sortField.FloatAt(i)
```

Comment:
Don't think we should use FloatAt, rather sort of each type.

Reply:
While the input won't be valid (so not commited as a go file),  it will still work to get you most of this, for example a new file in the /data folder called `generic_vector_less.go` and then contents like:

```
package data

//go:generate genny -in=$GOFILE -out=vector_less.gen.go gen "gen=uint8,uint16,uint32,uint64,int8,int16,int32,int64,float32,float64"

func (v *genVector) Less(i, j int) bool {
	return (*v)[i] < (*v)[j]
}
```

will work with `go generate generic_vector_less.go` run with the data folder. Then do something similar for the nullable vector (checking for nil, and sorting nil .. first?). And then hand written ones for bool and time (time as `Before` method which takes care of time sort (or maybe After?).

Finally the vector interface would get the Less method, and then can add a method to the Frame that calls that interface.

Reply:
https://github.com/grafana/grafana-plugin-sdk-go/blob/6807a8d3ceae94b2811a7fae3996f9c729568b5e/data/time_series.go#L556 for an example of sorting when there are nils

Reply:
@kylebrandt ```cannot compare (*v)[i].(*gen) < (*v)[j].(*gen) (operator < not defined for *gen)```

Is this possible with ordering operators?

Reply:
> @kylebrandt `cannot compare (*v)[i] < (*v)[j] (operator < not defined for *gen)`
> 
> Is this possible with ordering operators?

Yes and No. So what we currently use for code generation, `genny`, uses a system where the templates are valid go code. So if you try to do these comparisons the template will throw those errors, because it isn't valid go code.

That being said, running `go generate` (which triggers) genny does seem to still create a valid generated go code, even though the input isn't valid.

So basically my think it is it can't be hacked a little, but the templates for less in there own files and just make them readmes with the instructions, and code blocks.

So this would all be a bit hacky. So I'm thinking of it more as a way to bootstrap most of the methods, where you still will have to handwrite some because of `time` objects.